multinet/api/tests/test_workspace.py

```diff
@@ -169,22 +174,44 @@ def test_workspace_rest_delete_no_access(
 
 
 @pytest.mark.django_db
-@pytest.mark.parametrize('permission', [WorkspacePermission.owner, WorkspacePermission.maintainer])
 def test_workspace_rest_get_permissions(
     workspace: Workspace,
     user: User,
     user_factory: UserFactory,
     authenticated_api_client: APIClient,
-    permission: WorkspacePermission,
 ):
-    workspace.set_user_permission(user, permission)
+    workspace.set_user_permission(user, WorkspaceRoleChoice.MAINTAINER)
+    workspace.set_owner(user_factory())
+    create_users_with_permissions(user_factory, workspace)
+    maintainer_names = [maintainer.username for maintainer in workspace.maintainers]
+    writer_names = [writer.username for writer in workspace.writers]
+    reader_names = [reader.username for reader in workspace.readers]
+
+    r = authenticated_api_client.get(f'/api/workspaces/{workspace.name}/permissions/')
+    r_json = r.json()
+
+    assert r.status_code == 200
+    assert r_json['public'] == workspace.public
+    assert r_json['owner']['username'] == workspace.owner.username
 
-    if permission == WorkspacePermission.maintainer:
-        workspace.set_user_permission(user, permission)
-        workspace.set_owner(user_factory())
-    else:
-        workspace.set_owner(user)
+    for maintainer in r_json['maintainers']:
+        assert maintainer['username'] in maintainer_names
+
+    for writer in r_json['writers']:
+        assert writer['username'] in writer_names
 
+    for reader in r_json['readers']:
+        assert reader['username'] in reader_names
```

Comment:
Feel free to ignore, but this is what I tend to do when I need to assert something over an iterable (changed the var to `user` to keep the line short).

```suggestion
    assert all([user['username'] in maintainer_names for user in r_json['maintainers']])
    assert all([user['username'] in writer_names for user in r_json['writers']])
    assert all([user['username'] in reader_names for user in r_json['readers']])
```